ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/MinimalState.java

```diff
@@ -105,6 +105,14 @@ public void addBundle(IPluginModelBase model, boolean update) {
 		BundleDescription desc = model.getBundleDescription();
 		long bundleId = desc == null || !update ? -1 : desc.getBundleId();
 		try {
+			if (model.getInstallLocation() == null) {
```

Comment:
This expression should be extracted into an own variable in order to ensure it has the same value here and below, especially since we probably have a race-condition here.
Created https://github.com/eclipse-pde/eclipse.pde/pull/861 to address this.

Reply:
Good point, I hadn't thought of a race condition!

Thank you for reviewing and merging

Reply:
maybe we should make that property final, it seems odd that the install-location can be reset afterwards.

Reply:
hm, could be tricky since most of the implementing classes don't actually have it as a field but delegate the call to another object.
If you could you provide a PR for this, I'd be happy to review it :-)